### PR TITLE
addons: add screenshot and fanart support for repo

### PR DESCRIPTION
**Pull request by CvH**
This is partly untested because I have no idea how the used repo script looks like.

I added some example data that it can be easily tested - real fanart/screenshots need to be added to the addons (if nothing is provided or changed at the addons it still works - without the new feature).

The repo script has to be slightly changed (at least i think it is cleaner) to get changelog and icon from the repo/ folder (repo folder includes icon, changelog, fanart and the screenshots).

As I am absolutely no bash genius pls check if the script is good enough.

image

### CODE REVIEW

File: scripts/create_addon [hunk 1]
```diff
@@ -91,6 +91,13 @@ pack_addon() {
   if [ ! -z "$PKG_ADDON_NAME" ] ; then
     ADDON_NAME="$PKG_ADDON_NAME"
   fi
+  for f in $PKG_DIR/repo/screenshot-*.{jpg,png}; do if [ -f $f ];
```
Comment (MilhouseVH) [inline scripts/create_addon]: I'd definitely move the `if [ ]; then` to its own line, to stay consistent with existing code/style.

Comment (CvH) [inline scripts/create_addon]: fixed

Comment (MilhouseVH) [inline scripts/create_addon]: `skin.estuary` uses the `resources` folder for artwork assets (icons, fanart, screenshots): https://github.com/xbmc/xbmc/blob/master/addons/skin.estuary/addon.xml#L112-L123

Is there any reason not to do the same? I suppose there's the possibility of a naming clash, but then what's to say an addon might have a `repo` folder in future? In that case, should we use something very unlikely to clash, such as `.libreelec`?

Comment (CvH) [inline scripts/create_addon]: yes there is a clash - we have already the `resources` folder for the `settings.xml` and languages, I thought this is rather messy to name it `resources` - also it copies the whole content of the `resources` folder to the addon.zip
there is no need to name it `repo` - i had simply no better idea (maybe `le_repo` ...)

File: scripts/create_addon [hunk 2]
```diff
@@ -91,6 +91,14 @@ pack_addon() {
   if [ ! -z "$PKG_ADDON_NAME" ] ; then
     ADDON_NAME="$PKG_ADDON_NAME"
   fi
+  for f in $PKG_DIR/repo/screenshot-*.{jpg,png};
+    do if [ -f $f ];
```
Comment (lrusak) [inline scripts/create_addon]: `do` should be on previous line

File: scripts/create_addon [hunk 3]
```diff
@@ -91,6 +91,14 @@ pack_addon() {
...
+      then
```
Comment (lrusak) [inline scripts/create_addon]: `then` should be on the previous line

File: scripts/create_addon [hunk 4]
```diff
@@ -117,11 +126,23 @@ pack_addon() {
...
     if [ -f $ADDON_BUILD/$PKG_ADDON_ID/icon.png ]; then
-      cp $ADDON_BUILD/$PKG_ADDON_ID/icon.png $ADDON_INSTALL_DIR/icon.png
+      mkdir -p $ADDON_INSTALL_DIR/repo
+      cp $ADDON_BUILD/$PKG_ADDON_ID/icon.png $ADDON_INSTALL_DIR/repo/icon.png
     fi
+    if [ -f $PKG_DIR/repo/fanart.jpg ]; then
+      mkdir -p $ADDON_INSTALL_DIR/repo
+      cp $PKG_DIR/repo/fanart.jpg $ADDON_INSTALL_DIR/repo/fanart.jpg
+    fi
+    for f in $PKG_DIR/repo/screenshot-*.{jpg,png}; do if [ -f $f ];
```
Comment (lrusak) [inline scripts/create_addon]: if should be on another line

File: scripts/create_addon [hunk 5]
```diff
@@ -91,6 +91,14 @@ pack_addon() {
   if [ ! -z "$PKG_ADDON_NAME" ] ; then
     ADDON_NAME="$PKG_ADDON_NAME"
   fi
+  for f in $PKG_DIR/repo/screenshot-*.{jpg,png};
+    do if [ -f $f ];
+      then
+        f=$(echo $f | sed 's/.*\///')
```
Comment (lrusak) [inline scripts/create_addon]: here you can just `basename` instead of `sed`

**Comment by lrusak** [comment]
Can you breakup the changes to the existing add-ons into their own commits?

**Comment by CvH** [comment]
@lrusak yea this is just example data (I remove it before merge), the addons are getting their stuff as follow up pr (the script still works correctly without any changes at the current addons).

File: scripts/create_addon (hunk 5 above)
Comment (CvH) [inline scripts/create_addon]: any preference ?

Comment (MilhouseVH) [inline scripts/create_addon]: Use `basename` if all you want is the name of the file without the rest of the path.

**Comment by CvH** [comment]
@lrusak @chewitt and news ?

**Comment by lrusak** [comment]
why not name the folder resources rather than repo? I think this is how it is done for kodi yes?

**Comment by CvH** [comment]
We have already an resources folder for addons, also the content of that folder get copied to the addon.zip. This sounds rather messy to me in tradeoff of just the folder name.

**Comment by lrusak** [comment]
but `resources` is what they use upstream https://github.com/xbmc/xbmc/tree/master/addons/skin.estuary/resources
https://github.com/xbmc/xbmc/tree/master/addons/skin.estouchy/resources

**Comment by CvH** [comment]
@lrusak 
`if [ -f $PKG_DIR/repo/*.jpg ] || [ -f $PKG_DIR/repo/*.png ];` 
maybe this could simplified (`*.{jpg,png};` don't work)
besides that the script was changed to resources and some cleanup, works for me

image

File: scripts/create_addon [hunk 6]
```diff
@@ -119,9 +133,21 @@ pack_addon() {
...
+    if [ -f $ADDON_BUILD/$PKG_ADDON_ID/resources/icon.png ]; then
+      mkdir -p $ADDON_INSTALL_DIR/resources
+      cp $ADDON_BUILD/$PKG_ADDON_ID/resources/icon.png $ADDON_INSTALL_DIR/resources
     fi
+    if [ -f $ADDON_BUILD/$PKG_ADDON_ID/resources/fanart.jpg ]; then
+      mkdir -p $ADDON_INSTALL_DIR/resources
+      cp $ADDON_BUILD/$PKG_ADDON_ID/resources/fanart.jpg $ADDON_INSTALL_DIR/resources
+    fi
+
+    for f in $ADDON_BUILD/$PKG_ADDON_ID/resources/screenshot-*.{jpg,png}; do if [ -f $f ];
```
Comment (lrusak) [inline scripts/create_addon]: please use `if` on a new line, like `if [ -f "$" ]; then`

**Comment by lrusak** [comment]
@CvH I think you are overly complicating this

the screenshots should go in `$PKG_DIR/source/resources` as this folder is automatically copied to the addon dir https://github.com/LibreELEC/LibreELEC.tv/blob/master/scripts/create_addon#L70

then you just need to parse the screenshots like you already do.

**Comment by CvH** [comment]
@lrusak
the problem I tried to work around is that we used resources folder before kodi adapted the name for a different task, my goal was to separate by "logic" folders and keep the structure instead of throwing everything into a folder. The real "fix" would be to rename the resources folder of our addons to something else (create other problems like backporting ...). I have no problem with one of the possibilities :) Pls decide what you prefer and I change it that way.

**Comment by lrusak** [comment]
I would rather just dump it into the resources folder :)

**Comment by CvH** [comment]
@lrusak 
changed

**Comment by lrusak** [comment]
I'm ok with this now, we can look at moving the icon.png at a later time.

**Comment by lrusak** [comment]
@chewitt with this your scripts may need some tweaking so you may want to test this before merging.

**Comment by CvH** [comment]
added generic fanart picture for addon and squashed

**Comment by lrusak** [comment]
LGTM

**Comment by chewitt** [comment]
needs rebasing after changes to add news support

**Comment by CvH** [comment]
rebased and tested

**Comment by CvH** [comment]
the weekly ping

**Comment by lrusak** [comment]
@CvH can you update the date on the commit and I'll merge, thanks

**Pull request merged**